aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -1429,23 +1443,13 @@ TORCH_IMPL_FUNC(any_all_out)(const Tensor& self, const Tensor& result) {
   allany_impl<0>(self, result, {}, false, or_stub);
 }
 
-Tensor &amin_out(const Tensor& self, IntArrayRef dim, bool keepdim, Tensor& result) {
-  TORCH_CHECK(self.scalar_type() == result.scalar_type(), "Expected the dtype for input and out to match, but got ",
-              self.scalar_type(), " for input's dtype and ",  result.scalar_type(), " for out's dtype.");
-  if (self.numel() == 0) {
-    zero_numel_check_dims(self, dim, "amin()");
-  }
-
-  auto iter = make_reduction("amin", result, self, dim, keepdim, self.scalar_type());
+TORCH_IMPL_FUNC(amin_out) (const Tensor& self, IntArrayRef dim, bool keepdim, const Tensor& result) {
+  c10::MaybeOwned<Tensor> in = c10::MaybeOwned<Tensor>::borrowed(self);
```

Comment:
I actually don't think that you need the `MaybeOwned` here - you should just be able to directly pass `self` into `meta::make_reduction`.

The `MaybeOwned` call is in the `amax_out` structured kernel further down too - I think that was an accident (feel free to remove that in this PR too!). I left a comment on what `MaybeOwned` does and why I think it's unnecessary here, if you want to take a look: https://github.com/pytorch/pytorch/pull/72124/files#r801716554

Reply:
Hi, @bdhirsh!
Thanks for the share.
Sure I'll go through it and update that. Thanks

Reply:
That was informative. I got to know the reason behind it. Mentioning it here. Please let me know if I'm missing something. Thanks
1.  `MaybeOwned<Tensor>` is used for performance optimization situations. It is used in cases where we have reference to the tensor so as to permit differing values at the same storage. Here, `self` is an input provided by the user so we cannot guarantee, it's a reference or not. Therefore we are not using it here.
2. The example you shared there [link](https://github.com/pytorch/pytorch/blob/dd9517cc4a69c25c314a84b95992baa405bbcc77/aten/src/ATen/native/ReduceOps.cpp#L1467). Here, `argmax` and `argmin` points to the same memory location if `dim` is provided by the user. So, it makes sense to use `MaybeOwned<Tensor>::borrowed`. If not provided it uses `MaybeOwned<Tensor>::owned` by allocating new storage via reshaping.

Does that make sense? Thanks

Reply:
Actually I think I said the wrong thing in that previous comment 😅. There are some pretty good docs on `MaybeOwned` in the wiki here: https://pytorch.org/cppdocs/notes/maybe_owned.html.

I just looked at where `MaybeOwned` was originally added to argmin/max ([here](https://github.com/pytorch/pytorch/pull/58338/files)), and in that code you can see that we sometimes want to pass in `self` directly to `make_reduction` (in which case it's fine to pass a `const Tensor&`), but we sometimes want to pass in `self.reshape({-1})` to `make_reduction` (in which case we're getting out a new Tensor object, so we can't pass a reference.

Reply:
Got it! Thanks for the description, @bdhirsh :)